File: tools/split_code.js
const fs = require('fs');
const path = require('path');

const work_dir = 'c:\\Users\\hjroh\\OneDrive\\Desktop\\고탈겜_통합';
const game_code = fs.readFileSync('c:\\Users\\hjroh\\OneDrive\\Desktop\\고탈겜_1\\sketch.js', 'utf8');
const story_code = fs.readFileSync('c:\\Users\\hjroh\\OneDrive\\Desktop\\고탈겜_2\\sketch.js', 'utf8');

function removeFunction(code, funcName) {
    const startIdx = code.indexOf(`function ${funcName}(`);
    if (startIdx === -1) return code;

    let braceIdx = code.indexOf('{', startIdx);
    let braceCount = 1;
    let endIdx = braceIdx + 1;
    
    while (braceCount > 0 && endIdx < code.length) {
        if (code[endIdx] === '{') braceCount++;
        else if (code[endIdx] === '}') braceCount--;
        endIdx++;
    }
    
    return code.slice(0, startIdx) + code.slice(endIdx);
}

let game_code_clean = game_code;
['preload', 'setup', 'draw', 'keyPressed'].forEach(fn => {
    game_code_clean = removeFunction(game_code_clean, fn);
});

// Replace width and height with 600 and 800, ensuring they are not properties
game_code_clean = game_code_clean.replace(/(?<!\.)\bwidth\b/g, '600');
game_code_clean = game_code_clean.replace(/(?<!\.)\bheight\b/g, '800');

let story_code_clean = story_code;
['preload', 'setup', 'draw', 'mousePressed', 'keyPressed', 'windowResized'].forEach(fn => {
    story_code_clean = removeFunction(story_code_clean, fn);
});

fs.writeFileSync(path.join(work_dir, 'game.js'), game_code_clean, 'utf8');
fs.writeFileSync(path.join(work_dir, 'story.js'), story_code_clean, 'utf8');

console.log("Split and replace completed via Node.");

File: tools/split_code.py
import os

work_dir = r"c:\Users\hjroh\OneDrive\Desktop\고탈겜_통합"

# Read game sketch
with open(os.path.join(work_dir, "sketch.js"), "r", encoding="utf-8") as f:
    game_code = f.read()

# Read story sketch 
# Wait, original story sketch is not easily identifiable if we overwrote it. 
# Let's read from original folders.
with open(r"c:\Users\hjroh\OneDrive\Desktop\고탈겜_1\sketch.js", "r", encoding="utf-8") as f:
    game_code = f.read()

with open(r"c:\Users\hjroh\OneDrive\Desktop\고탈겜_2\sketch.js", "r", encoding="utf-8") as f:
    story_code = f.read()

# Extract globals and classes from game_code
# game_code has globals at the top, then preload, setup, draw, keyPressed, then GAME FLOW, WORLD GENERATION, SCENES, BACKGROUND DRAWING, CAT DRAWING, CLASSES
# Let's just output game_code minus standard p5 functions.
import re

def remove_function(code, func_name):
    pattern = r"function\s+" + func_name + r"\s*\([\s\S]*?\n}\n"
    # This naive regex might fail with nested braces. 
    # Let's do a brace matching extraction.
    start_idx = code.find(f"function {func_name}(")
    if start_idx == -1: return code
    
    # Find the opening brace
    brace_idx = code.find("{", start_idx)
    brace_count = 1
    end_idx = brace_idx + 1
    while brace_count > 0 and end_idx < len(code):
        if code[end_idx] == '{': brace_count += 1
        elif code[end_idx] == '}': brace_count -= 1
        end_idx += 1
        
    return code[:start_idx] + code[end_idx:]

game_code_clean = game_code
for fn in ["preload", "setup", "draw", "keyPressed"]:
    game_code_clean = remove_function(game_code_clean, fn)

story_code_clean = story_code
for fn in ["preload", "setup", "draw", "mousePressed", "keyPressed", "windowResized"]:
    story_code_clean = remove_function(story_code_clean, fn)

with open(os.path.join(work_dir, "game.js"), "w", encoding="utf-8") as f:
    f.write(game_code_clean)

with open(os.path.join(work_dir, "story.js"), "w", encoding="utf-8") as f:
    f.write(story_code_clean)

print("Split completed.")
